lit_llama/adapter_v2.py

```diff
@@ -26,20 +26,34 @@ def adapter_v2_state_from_state_dict(state_dict: dict) -> dict:
 
 
 def adapter_v2_new_forward(self, input: Tensor) -> Tensor:
+    weight = self.weight
+
+    try:
+        from lit_llama.quantization import Linear8bitLt
+        if isinstance(self, Linear8bitLt):
+            weight = self.dequantize(input.dtype)
+    except:
+        None
     return self.adapter_scale * (
-        F.linear(input, self.weight, self.bias) + self.adapter_bias
+        F.linear(input, weight, self.bias) + self.adapter_bias
     )
 
 
-def adapter_v2_linear_with_bias_and_scale(layer):
-    layer.adapter_bias = torch.nn.Parameter(torch.zeros(layer.weight.shape[0]), requires_grad=True)
-    layer.adapter_scale = torch.nn.Parameter(torch.ones(layer.weight.shape[0]), requires_grad=True)
+def adapter_v2_linear_with_bias_and_scale(layer, dtype=None, quantize=False):
+    weight = layer.weight
+
+    if dtype is not None and quantize:
+        from lit_llama.quantization import Linear8bitLt
+        if isinstance(layer, Linear8bitLt):
```

Comment:
If the snippet above uses a try-catch, wouldn't you want it here too?

Reply:
@carmocca Good point. And I just remember now why I didn't do it. I had some issues here with that.

So, some people may not be able to install `bitsandbytes`, and that shouldn't prevent people from using the adapter method without quantization. So, that's why added the `quantize` argument here. But if someone using the quantization flag, which sets `quantize=True` here AND `bitsandbytes` can not be imported. It SHOULD fail, because otherwise it would run without quantization which is not what's intended when someone uses `--quantize`. 

Now, in this case above where I used the try-except, I failed making it work with the `quantize` method because I am overriding the default forward method, and I don't think it's easily possible to add that as an argument. I am actually not sure about that and would need some help here. 

I think we actually want to remove the try-except above somehow as this is stupid and expensive if it has to fail to import something every time a forward call happens. Any ideas?

<img width="724" alt="Screenshot 2023-06-07 at 4 24 53 PM" src="https://github.com/Lightning-AI/lit-llama/assets/5618407/9d3c30c4-d16a-406b-8cc3-ed7240ae74cd">

Reply:
I see (nice image!).

We could do this with a `partial`: `partial(adapter_v2_new_forward, quantize=quantize)`